pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -188,7 +188,7 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 		obj.CPUCFSQuota = utilpointer.BoolPtr(true)
 	}
 	if obj.CPUCFSQuotaPeriod == nil {
-		obj.CPUCFSQuotaPeriod = &metav1.Duration{Duration: 100 * time.Millisecond}
+		obj.CPUCFSQuotaPeriod = &metav1.Duration{Duration: 100 * time.Microsecond}
```

Comment:
Changing defaults is generally not allowed and is considered an incompatible change.

It looks like the current 100ms default has been in place since 1.12 (https://github.com/kubernetes/kubernetes/pull/63437)

I don't have context to know what the implications of making the default 1000x shorter are on all existing clusters. cc @kubernetes/sig-node-api-reviews

Reply:
To clarify, by default clusters still use [the 100us value](https://github.com/kubernetes/kubernetes/blob/37761a3/pkg/kubelet/cm/helpers_linux.go#L48-L67) which is [in sync with Linux](https://www.kernel.org/doc/html/latest/scheduler/sched-bwc.html#management).

Enabling `CustomCPUCFSQuotaPeriod` flag alongside providing a custom `cpuCFSQuotaPeriod` value leaves the cluster owner with the selected value.

Only clusters with `CustomCPUCFSQuotaPeriod` enabled but without `cpuCFSQuotaPeriod` would be affected and receive 1000x shorter value.

I'm as upset as you are about this change, but I think it has to be done. The improper default value for `cpuCFSQuotaPeriod` introduces even more confusion to an already complicated topic (see #51135 for context).

Reply:
It does look that if the `CustomCPUCFSQuotaPeriod` (still alpha) feature gate is not enabled, then the period will be read from the constant QuotaPeriod and ignore the kubelet config value for `CPUCFSQuotaPeriod` - xref: https://github.com/kubernetes/kubernetes/blob/f3d90aef8d4a0a6934a4a51705c241a86f66b00c/pkg/kubelet/cm/helpers_linux.go#L65-L67

It sounds like the problem is the original default here was invalid in the first place -- it should have been the same value as the linux kernel. And the original default confused `ms` to mean millisecond instead of microseconds :). The value should be microseconds, quoting from kernel docs - "cpu.cfs_quota_us: run-time replenished within a period (in microseconds)"

Additionally, we should consider if validation should be added for the kubelet config, enabling the `CustomCPUCFSQuotaPeriod` feature gate without specifying a `CPUCFSQuotaPeriod` should be invalid...

Reply:
@bobbypage yes that would be a great solution, then even if people have this wrong configuration they would get a notification before going silently in production with this.

Reply:
I've altered this PR to include only changing the default value, and I updated the code and PR description (so please re-review).

Harmless documentation update, which just reflects the real state of code more precisely, was moved to #111554.

I might have an opinion about restricting users from enabling `CustomCPUCFSQuotaPeriod` but not setting `CPUCFSQuotaPeriod`, but I don't have arguments to support it. Tell me what to do, and I'll alter the PR as you say. If the vote is pro the change, please point me to any option which works in a similar way as I'm not familiar with that project code and the only way for me to do it right is to copy same code from another place.

Reply:
hmm, thinking about it some more, the issue with adding the validation is if any users will currently enable the feature gate and also don't set a `CPUCFSQuotaPeriod` then their kubelet will not start up (as this would be a breaking change). Additionally, if the feature gate would ever be promoted to beta and it would be turned on by default we would not want the validation to fail if the a CPUCFSQuotaPeriod is not set. So thinking about it some more, I think we should just focus on making the default correct.

Reply:
What if we first make it fail and 3 versions later make the default correct?

Reply:
Code should be written in a way that it will be correct when the feature gate graduates to GA, is unconditionally enabled, and removed in the future. Requiring a custom value if the feature gate is enabled doesn't match that.

Changing the default while the feature is in the alpha state seems reasonable and sufficient to me

Reply:
Thanks @liggitt